### ISSUE TITLE
Bug caused by creating custom layer

### ISSUE THREAD

**Issue by hoangcuong2011** [tensorflow/tensorflow#39496]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS but it should work with Ubuntu as well.
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.15
- Python version: 3.6
- GPU model and memory: CPU

I found a bug related to creating custom layer in TF. Here is a short code to replicate::

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import tensorflow as tf

class CustomModel(tf.keras.layers.Layer):
    #this class is for source sequence
    def __init__(self,):
        super(CustomModel, self).__init__()

    def build(self, input_shape):
        self.kernel = self.add_weight(shape=(32, 512, 512),
                                      initializer=tf.keras.initializers.glorot_uniform(seed=1),
                                      trainable=True) #dumb
    def call(self, inputs):
        return inputs[0]

def main():
    def create_model(source_vocab, target_vocab, relationship_vocab):
        source = tf.keras.layers.Input(dtype='int32', shape=(64,), name='source')
        target = tf.keras.layers.Input(dtype='int32', shape=(64,), name='target')
        relationship = tf.keras.layers.Input(dtype='int32', shape=(1,), name='relationship')
        embedding_source = tf.keras.layers.Embedding(512, 512, input_length=64)(source)
        embedding_target = tf.keras.layers.Embedding(512, 512, input_length=64)(target)
        final_layer = CustomModel()([relationship, embedding_source, embedding_target])
        model = tf.keras.models.Model(inputs=[source, target, relationship], outputs=final_layer)
        return model
    model = create_model(1000, 1000, 500)
    print(model.summary())

if __name__ == '__main__':
    main()

```
I believe this is a bug so I posted it here so that you can find and fix it.
Meanwhile one workaround I found is to change the order of parameters. Below you can find the solution:

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import tensorflow as tf

class CustomModel(tf.keras.layers.Layer):
    #this class is for source sequence
    def __init__(self,):
        super(CustomModel, self).__init__()

    def build(self, input_shape):
        self.kernel = self.add_weight(shape=(32, 512, 512),
                                      initializer=tf.keras.initializers.glorot_uniform(seed=1),
                                      trainable=True) #dumb
    def call(self, inputs):
        return inputs[0]

def main():
    def create_model(source_vocab, target_vocab, relationship_vocab):
        source = tf.keras.layers.Input(dtype='int32', shape=(64,), name='source')
        target = tf.keras.layers.Input(dtype='int32', shape=(64,), name='target')
        relationship = tf.keras.layers.Input(dtype='int32', shape=(1,), name='relationship')
        embedding_source = tf.keras.layers.Embedding(512, 512, input_length=64)(source)
        embedding_target = tf.keras.layers.Embedding(512, 512, input_length=64)(target)
        final_layer = CustomModel()([embedding_source, embedding_target, relationship])
        model = tf.keras.models.Model(inputs=[source, target, relationship], outputs=final_layer)
        return model
    model = create_model(1000, 1000, 500)
    print(model.summary())

if __name__ == '__main__':
    main()

```

**Comment by Saduf2019** [comment]
@hoangcuong2011 
I ran the code shared by you, please find the [gist here](https://colab.sandbox.google.com/gist/Saduf2019/7a7550923f0b89778291f17d2ac356b1/untitled181.ipynb).
Could you tell us what i the error as we can see an output with out any error as per the gist.
I ran both the codes shared and see the same output.

**Comment by Saduf2019** [comment]
Is there any particular reason for using an older version of tf, could you please upgrade to a later version.

**Comment by hoangcuong2011** [comment]
@Saduf2019: hello - what you run was with TF 2.1. It seems works on this version indeed.
Nonetheless it does not work for TF 1.15. (see the [gist](https://colab.research.google.com/drive/1h5eG4tqmewTFdVA_WZrkwMOPrQ3oJIdx?usp=sharing)).
I understand this bug is only with TF 1.15 so it is with a low priority from your side to fix (perhaps you will leave it as it is). I just wanted to raise your attention on this. I myself still use this version simply because some code I wrote could not run with TF 2.1. Nonetheless I know that it is just a matter of time and I need to move to TF. 2x one day, though.

**Comment by Saduf2019** [comment]
@hoangcuong2011 
We are not going to have any bug fixes on 1.15, please upgrade to 2.x.

**Comment by Saduf2019** [comment]
Fixed in 2.x, masking this as closed
